web-extension/test/admin-ns.spec.js

```diff
@@ -572,4 +591,75 @@ describe('admin-ns', () => {
       expect(exportKey.result.secretKey).not.toBeNull()
     })
   })
+
+  describe('admin.rename_key', () => {
+    const passphrase = 'foo'
+    let admin
+    let key
+    beforeEach(async () => {
+      admin = await createAdmin({ passphrase })
+
+      const generateRecoveryPhrase = await admin.onrequest(REQ_GENERATE_RECOVERY_PHRASE(1))
+      expect(generateRecoveryPhrase.error).toBeUndefined()
+
+      const importWallet = await admin.onrequest(REQ_IMPORT_WALLET(2, generateRecoveryPhrase.result.recoveryPhrase))
+      expect(importWallet.error).toBeUndefined()
+
+      const generateKey = await admin.onrequest(REQ_GENERATE_KEY(3))
+      expect(generateKey.error).toBeUndefined()
+
+      key = generateKey.result
+    })
+
+    afterEach(() => {
+      admin = null
+      key = null
+    })
+
+    it('should not rename key with wrong public key', async () => {
```

Comment:
also are there any special characters or things that should not be allowed in the key naming? Or are we keeping validation loose on this.

Reply:
Added tests for this :+1:

Consensus seems to be that since the storage doesn't care what name you give your keys, we will keep validation loose. Right now this endpoint is also only available to the user themselves